File: src/dataset/cpp/08-raindrops_exercism-ARPIT73881/openAI/ChatGPT4_raindrops_v2.cpp
#include "raindrops.h"
#include <string>

namespace raindrops {

    std::string convert(int number) {
        if (number % 3 != 0 && number % 5 != 0 && number % 7 != 0) {
            return std::to_string(number);
        }

        std::string result;
        if (number % 3 == 0) result += "Pling";
        if (number % 5 == 0) result += "Plang";
        if (number % 7 == 0) result += "Plong";

        return result;
    }

}  // namespace raindrops

File: src/dataset/typescript/raindrops_exercism-wackerow/openAI/ChatGPT4_raindrops_v2.ts
export default class Raindrops {
  public convert(num: number): string {
    const sounds = [
      { factor: 3, sound: 'Pling' },
      { factor: 5, sound: 'Plang' },
      { factor: 7, sound: 'Plong' },
    ];
    let raindrop = sounds.reduce((acc, { factor, sound }) => 
      num % factor === 0 ? acc + sound : acc, 
      ''
    );
    return raindrop || num.toString();
  }
}